leetcode/17_letter_combinations_of_a_phone_number.py

```diff
@@ -0,0 +1,101 @@
+from typing import List
+from itertools import product
+
+class Solution:
+    def letterCombinations(self, digits: str) -> List[str]:
+        """Generates all possible letter combinations that the given phone number digits
+        could represent.
+
+        Args:
+            digits (str): A string containing digits from '2' to '9'.
+
+        Returns:
+            List[str]: A list of all possible letter combinations.
+        """
+        if not digits:
+            return []
+
+        dic = {
+            "2": "abc", "3": "def", "4": "ghi", "5": "jkl",
+            "6": "mno", "7": "pqrs", "8": "tuv", "9": "wxyz"
+        }
+
+        groups = []
+        for digit in digits:
+            groups.append(dic[digit])
+
+        combinations = product(*groups)
+
+        result = []
+        for combination in combinations:
+            result.append("".join(combination))
+
+        return result
+
+    def recursiveletterCombinations(self, digits: str) -> List[str]:
+        """Generates all possible letter combinations that the given phone number digits could
+        represent using a recursive backtracking approach.
+
+        Args:
+            digits (str): A string containing digits from '2' to '9'.
+
+        Returns:
+            List[str]: A list of all possible letter combinations.
+        """
+        if not digits:
+            return []
+
+        ans, sol = [], []
+
+        dic = {
+            "2": "abc", "3": "def", "4": "ghi", "5": "jkl",
+            "6": "mno", "7": "pqrs", "8": "tuv", "9": "wxyz"
+        }
+
+        n = len(digits)
+
+        def backtrack(i):
```

Comment:
Rather than use `backtrack(i)` as a helper function. 

What's preventing you from simply using `recursiveletterCombinations` itself in the recursive call?